supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -415,36 +477,83 @@ class TesAsyncBackendJobExecutionActor(override val standardParams: StandardAsyn
       }
     } yield statusWithLog
 
-  private def queryStatusAsync(handle: StandardAsyncPendingExecutionHandle): Future[TesRunStatus] =
-    makeRequest[MinimalTaskView](HttpRequest(uri = s"$tesEndpoint/${handle.pendingJob.jobId}?view=MINIMAL")) map {
-      response =>
-        val state = response.state
-        state match {
-          case s if s.contains("COMPLETE") =>
-            jobLogger.info(s"Job ${handle.pendingJob.jobId} is complete")
-            Complete
+  private def queryStatusAndCostData(handle: StandardAsyncPendingExecutionHandle,
+                                     fetchCostData: Boolean
+  ): Future[TesRunStatus] =
+    if (fetchCostData) {
+      makeRequest[Task](HttpRequest(uri = s"$tesEndpoint/${handle.pendingJob.jobId}?view=FULL")) map { response =>
+        val fetchedCostData = for {
+          responseLogs <- response.logs
+          startTime <- responseLogs.head.start_time
+          vmCost <- responseLogs.head.metadata.flatMap(_.get("vm_price_per_hour_usd"))
```

Comment:
I think `head` will throw the same way as `get` if the list is empty - better to use `headOption`, which will return `None` when the list is empty.